_posts/2017-09-05-Rust-2017-Survey-Results.md

```diff
@@ -0,0 +1,126 @@
+
+![Chart: 91.5% Linux, 46.7% Windows, 38.2% macOS, 16.8% embedded, 13.2% WebAssembly and asm.js, 9.9% Android, 8.9% BSD-variant, 5.3% Apple iOS](/images/2017-09-05-Rust-2017-Survey-Results/platforms_targetted.png)
+
+Next, we asked what platforms people were targeting with their Rust projects.  While we see a similar representation of desktop OSes here, we also see a growing range of other targets.  Android and iOS are at healthy 9.9% and 5.3% respectively, both almost **10x larger** than last year's percentages.  Embedded also has had substantial growth since last year's single-digit percentage.  As a whole, cross-compilation has grown considerably since this time last year.
+
+![Chart: 45.8% vim, 33.8% vscode, 16.1% intellij, 15.7% atom, 15.4% emacs, 12.2% sublime, 1.5% eclipse, 1.5% visual studio](/images/2017-09-05-Rust-2017-Survey-Results/editors.png)
+
+Among editors, vim remains king, though we see healthy growth in VSCode adoption at 34.1% (up from last year's 3.8%).  This growth no doubt has been helped by VSCode being one of the first platforms to get support for the [Rust Language Server](https://github.com/rust-lang-nursery/rls).
+
+![Chart: 4.4% full-time, 16.6% part-time, 2.9% no but company uses Rust, 57.6% no, 2.4% not sure, 16.1% not applicable](/images/2017-09-05-Rust-2017-Survey-Results/use_rust_at_work.png "4.4% full-time, 16.6% part-time, 2.9% no but company uses Rust, 57.6% no, 2.4% not sure, 16.1% not applicable")
+*(hover for more info)*
+
+Rust in the workplace has also continued to grow.  This year's **4.4% full-time** and **16.6% part-time** Rust workers show a tick up from last year's 3.7% full-time and 16.1% part-time.
+
+![Two charts part-time: 2016: 29.6% less than 1000 lines, 55.3% 1000 to 10000 lines, 14.5% 10000 to 100000 lines, 0.6% 100000 lines. 2017: 18.9% less than 1000 lines, 56% 1000 to 10000 lines, 23.1% 10000 to 100000 lines, 2% more than 100000 lines](/images/2017-09-05-Rust-2017-Survey-Results/part_time.png)
+
+Users who use Rust **part-time** in their companies showed a growth in larger projects since last year, with the medium- and large-scale projects taking up more percentage of total projects this time around.
+
+![Two charts full-time: 2016: 4.4% less than 1000 lines, 42.6% 1000 to 10000 lines, 39.7% 10000 to 100000 lines, 13.2% more than 100000 lines. 2017: 1.9% less than 1000 lines, 27.9% 1000 to 10000 lines, 52.6% 10000 to 100000 lines, 17.5% more than 100000 lines](/images/2017-09-05-Rust-2017-Survey-Results/full_time.png)
+
+Likewise, **full-time** Rust commercial users saw medium- and large-scale projects grow to taking a larger part of the pie, with projects over 100,000 lines of code making up almost 18% of the all full-time commercial respondents, and a large shift in the 10,000-100,000 lines range from 39.7% up to **52.6%**.
```

Comment:
Hmm, I assume the 100,000+ people are all Mozilla employees (unless someone is doing a giant Rust project we don't know about)? Not sure how much attention we should bring to that.

Reply:
From a quick look, the vast majority aren't Mozilla employees (as best as we can tell).  We know of some pretty large external Rust projects, but it does make me wonder if we shouldn't reach out to those people who say it's okay to and see what they're doing...

Reply:
Huh, interesting, we should def. reach out if we can. Perhaps the total project size is > 100,000 lines rather than > 100,000 lines of Rust?